scanner/src/main/kotlin/provenance/NestedProvenanceResolver.kt

```diff
@@ -38,7 +36,7 @@ interface NestedProvenanceResolver {
      * [NestedProvenance] always contains only the provided [ArtifactProvenance]. For a [RepositoryProvenance] the
      * resolver looks for nested repositories, for example Git submodules or Mercurial subrepositories.
      */
-    fun resolveNestedProvenance(provenance: KnownProvenance): NestedProvenance
+    suspend fun resolveNestedProvenance(provenance: KnownProvenance): NestedProvenance
```

Comment:
Can callers like

https://github.com/oss-review-toolkit/ort/blob/74557ba1428f578d1456535435591292714c469d/scanner/src/main/kotlin/Scanner.kt#L272-L276

now be simplified?

Reply:
No, the `async` is used here to run multiple resolutions in parallel, the `runCatching` to catch exceptions, and the `provenance to` to make the `provenance` available for later code.